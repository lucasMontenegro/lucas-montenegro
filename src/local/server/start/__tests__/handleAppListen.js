const mockFn = jest.fn()
const mockGlobals = require("local/utils/mockGlobals")
const globals = {
  process: { env: {}, pid: `'process.pid'` },
  console: { log: (...args) => mockFn(`console.log`, args) },
}
const oldGlobals = mockGlobals(globals)
const handleAppListen = require("local/server/start/handleAppListen")
mockGlobals(oldGlobals)
describe(`local/server/start/handleAppListen`, () => {
  afterEach(() => mockFn.mockClear())
  test.each([
    [`production`],
    [`foo`],
  ])(`should log (NODE_ENV %s)`, NODE_ENV => {
    globals.process.env.NODE_ENV = NODE_ENV
    handleAppListen(`'PORT'`)()
    expect(mockFn.mock.calls).toMatchSnapshot()
  })
})