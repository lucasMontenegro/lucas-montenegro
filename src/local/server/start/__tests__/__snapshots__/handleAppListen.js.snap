// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`local/server/start/handleAppListen should log (NODE_ENV foo) 1`] = `
Array [
  Array [
    "console.log",
    Array [
      "Node dev server: listening on port 'PORT'",
    ],
  ],
]
`;

exports[`local/server/start/handleAppListen should log (NODE_ENV production) 1`] = `
Array [
  Array [
    "console.log",
    Array [
      "Node cluster worker 'process.pid': listening on port 'PORT'",
    ],
  ],
]
`;
